Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -0,0 +1,214 @@
+     */
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    int is_boundary_PEC (amrex::GpuArray<int, 3> const& fboundary, int dir) {
+        return int( fboundary[dir] == FieldBoundaryType::PEC);
+    }
+
+    /**
+     * \brief Sets the electric field value tangential to the PEC boundary to zero. The
+     *        tangential Efield components in the guard cells outside the
+     *        domain boundary are set equal and opposite to the field in the valid cells
+     *        at their mirrored locations. The normal Efield components in the guard cells
+     *        are set equal to the field in the valid cells at their mirrored locations.
+     *        The number or depth of guard cells updated is equal to the shape factor of
+     *        particles in each dimension.
+     *        For corner cells with mixed boundaries, the mirror location could be outside
+     *        valid region, while still ensuring PEC condition is maintained across the
+     *        PEC boundary, and the necessary sign change is accounted for depending on
+     *        if the component, icomp, is tangential or normal to the PEC boundary.
+     *
+     * \param[in] icomp        component of the Efield being updated
+     *                         (0=x, 1=y, 2=z in Cartesian)
+     *                         (0=r, 1=theta, 2=z in RZ)
+     * \param[in] dom_lo       index value of the lower domain boundary (cell-centered)
+     * \param[in] dom_hi       index value of the higher domain boundary (cell-centered)
+     * \param[in] ijk_vec      indices along the x(i), y(j), z(k) of Efield Array4
+     * \param[in] n            index of the MultiFab component being updated
+     * \param[in] Efield       field data to be updated if (ijk) is at the boundary or a guard cell
+     * \param[in] stag         staggering of the field data being updated.
+     * \param[in] fbndry_lo    Field boundary type at the lower boundaries
+     * \param[in] fbndry_hi    Field boundary type at the upper boundaries
+     */
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    void SetEfieldOnPEC (const int icomp, const amrex::IntVect & dom_lo,
+                                const amrex::IntVect &dom_hi,
+                                const amrex::IntVect &ijk_vec, const int n,
+                                amrex::Array4<amrex::Real> const& Efield,
+                                const amrex::IntVect& stag,
+                                amrex::GpuArray<int, 3> const& fbndry_lo,
+                                amrex::GpuArray<int, 3> const& fbndry_hi )
+    {
+        // Tangential Efield componentes in guard cells set equal and opposite to cells
+        // in the mirror locations across the PEC boundary, whereas normal E-field
+        // components are set equal to values in the mirror locations across the PEC
+        // boundary. Here we just initialize it.
+        amrex::IntVect ijk_mirror = ijk_vec;
+        int OnPECBoundary = 0;
```

Comment:
A minor detail is that these could be `bool` variables, unless there is some reason that bools shouldn't be used in kernel code.

Reply:
done :) Thank you